dvc/api/experiments.py

```diff
@@ -69,3 +72,86 @@ def exp_save(
         return repo.experiments.save(
             name=name, force=force, include_untracked=include_untracked
         )
+
+
+def _postprocess(exp_rows):
+    for exp_row in exp_rows:
+        for k, v in exp_row.items():
+            if isinstance(v, Text):
+                v_str = str(v)
+                try:
+                    exp_row[k] = float(v_str)
+                except ValueError:
+                    exp_row[k] = v_str
+    return exp_rows
+
+
+def exp_show(
+    repo: Optional[str] = None,
+    revs: Optional[Union[str, List[str]]] = None,
+    num: int = 1,
+    hide_queued: bool = False,
+    hide_failed: bool = False,
+    sha: bool = False,
+    param_deps: bool = False,
+    force: bool = False,
+) -> List[Dict]:
+    """Get DVC experiments tracked in `repo`.
+
+    Without arguments, this function will retrieve all experiments derived from
+    the Git `HEAD`.
+
+    See the options below to customize the experiments retrieved.
+
+    Args:
+        repo (str, optional): location of the DVC repository.
+            Defaults to the current project (found by walking up from the
+            current working directory tree).
+            It can be a URL or a file system path.
+            Both HTTP and SSH protocols are supported for online Git repos
+            (e.g. [user@]server:project.git).
+        revs (Union[str, List[str]], optional): Git revision(s) (e.g. branch,
+            tag, SHA commit) to use as a reference point to start listing
+            experiments.
+            Defaults to `None`, which will use `HEAD` as starting point.
+        num (int, optional): show experiments from the last `num` commits
+            (first parents) starting from the `revs` baseline.
+            Give a negative value to include all first-parent commits (similar
+            to `git log -n`).
+            Defaults to 1.
+        hide_queued (bool, optional): hide experiments that are queued for
+            execution.
+            Defaults to `False`.
+        hide_failed (bool, optional): hide experiments that have failed.
+        sha (bool, optional): show the Git commit SHAs of the experiments
+            instead of branch, tag, or experiment names.
+            Defaults to `False`.
+        param_deps (bool, optional): include only parameters that are stage
+            dependencies.
+            Defaults to `False`.
```

Comment:
Sorry @daavoo, I should have reviewed this closer before merging. Nothing severe, but I have minor questions when you have a chance:

* Why did you choose to include these options to hide rows or columns but not others like `only-changed`?
* What's the point of including `sha` since there's already a `rev` column?

Reply:
> Why did you choose to include these options to hide rows or columns but not others like only-changed?

Because these are applied at the "low level" collection and the others are applied on the UI. I will open P.R. dropping them, though

> What's the point of including sha since there's already a rev column?

There is no point.